docs/features/kubernetes/authentication.md

```diff
@@ -28,6 +28,28 @@ The providers available as server side are:
 - `localKubectlProxy`
 - `serviceAccount`
 
+### Azure
+
+The Azure server side authentication provider works by authenticating on the server with
+the Azure CLI `az login`, meaning any user that can access the backstage app will have
+the same level of access to the cluster in the Kubernetes plugin, to put an example, even
+guest users would have access to the cluster resources.
```

Comment:
This is actually true for any server side auth provider, so maybe call it out in the above section rather than here?

Reply:
I only got to test Azure clusters with the guest login, so I was not sure if it was indeed the case for all server-side providers, I'll move & reword this to the server-side section directly.